addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -166,7 +174,34 @@ public void run() {
                     logger.error("An exception occurred while executing '{}' : '{}'",
                             new Object[] { commandLine.toString(), e.getMessage() });
                     updateState(RUN, OnOffType.OFF);
-                    updateState(OUTPUT, new StringType(e.getMessage()));
+
+                    if (e.getMessage().equals("OPEN")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.OPEN);
+                    }
+
+                    if (e.getMessage().equals("CLOSED")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.CLOSED);
+                    }
+
+                    if (e.getMessage().equals("ON")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.ON);
+                        updateState(SWITCH_OUTPUT, OnOffType.ON);
+                    }
+
+                    if (e.getMessage().equals("OFF")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.OFF);
+                        updateState(SWITCH_OUTPUT, OnOffType.OFF);
+                    }
+
+                    if (e.getMessage().equals("UP")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.UP);
+                    }
+
+                    if (e.getMessage().equals("DOWN")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.DOWN);
+                    }
+
+                    updateState(STRING_OUTPUT, new StringType(e.getMessage()));
```

Comment:
If I understand this code correctly, the output data type is casted to the different item data types supported by OH2. But I can't see castings to the datatype Percent which is used by the Rollershutter item for example. Is this conversion missing or is it done somewhere else?